marker/v2/schema/groups/page.py

```diff
@@ -42,6 +42,4 @@ def add_full_block(self, block: Block) -> Block:
         return block
 
     def get_block(self, block_id: BlockId) -> Block | None:
-        for block in self.children:
-            if block.id == block_id:
-                return block
+        return self.children[block_id.block_id]
```

Comment:
We should add an assert to make sure the ids match, just in case something gets inserted out of order

Reply:
Makes sense! Pushed!